internal/provider/resources/resource_cluster_workspace_authorization.go

```diff
@@ -0,0 +1,381 @@
+	ctx context.Context,
+	req resource.SchemaRequest,
+	resp *resource.SchemaResponse,
+) {
+	resp.Schema = schema.Schema{
+		// This description is used by the documentation generator and the language server.
+		MarkdownDescription: "Cluster workspace authorization resource.",
+		Attributes:          schemas.ResourceClusterWorkspaceAuthorizationSchemaAttributes(),
+	}
+}
+
+func (r *ClusterWorkspaceAuthorizationResource) Configure(
+	ctx context.Context,
+	req resource.ConfigureRequest,
+	resp *resource.ConfigureResponse,
+) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	apiClients, ok := req.ProviderData.(models.ApiClientsModel)
+	if !ok {
+		utils.ResourceApiClientConfigureError(ctx, req, resp)
+		return
+	}
+
+	r.platformClient = apiClients.PlatformClient
+	r.organizationId = apiClients.OrganizationId
+}
+
+func (r *ClusterWorkspaceAuthorizationResource) Create(
+	ctx context.Context,
+	req resource.CreateRequest,
+	resp *resource.CreateResponse,
+) {
+	var data models.ClusterWorkspaceAuthorizationResource
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var diags diag.Diagnostics
+	var updateClusterRequest platform.UpdateClusterRequest
+	updateDedicatedClusterRequest := platform.UpdateDedicatedClusterRequest{}
+
+	// workspaceIds
+	if !data.WorkspaceIds.IsNull() {
+		workspaceIds, diags := utils.TypesSetToStringSlice(ctx, data.WorkspaceIds)
+		updateDedicatedClusterRequest.WorkspaceIds = &workspaceIds
+		if diags.HasError() {
+			resp.Diagnostics.Append(diags...)
+			return
+		}
+	}
+
+	err := updateClusterRequest.FromUpdateDedicatedClusterRequest(updateDedicatedClusterRequest)
+	if err != nil {
+		tflog.Error(ctx, fmt.Sprintf("failed to update cluster error: %v", err))
+		resp.Diagnostics.AddError(
+			"Client Error",
+			fmt.Sprintf("Unable to update cluster request body, got error: %s", err),
+		)
+		return
+	}
+
+	cluster, err := r.platformClient.UpdateClusterWithResponse(ctx, r.organizationId, data.ClusterId.ValueString(), updateClusterRequest)
+	if err != nil {
+		tflog.Error(ctx, "failed to update cluster", map[string]interface{}{"error": err})
+		resp.Diagnostics.AddError(
+			"Client Error",
+			fmt.Sprintf("Unable to update cluster, got error: %s", err),
+		)
+		return
+	}
+	_, diagnostic := clients.NormalizeAPIError(ctx, cluster.HTTPResponse, cluster.Body)
+	if diagnostic != nil {
+		resp.Diagnostics.Append(diagnostic)
+		return
+	}
+
+	// Wait for the cluster to be updated (or fail)
+	stateConf := &retry.StateChangeConf{
+		Pending:    []string{string(platform.ClusterStatusCREATING), string(platform.ClusterStatusUPDATING)},
+		Target:     []string{string(platform.ClusterStatusCREATED), string(platform.ClusterStatusUPDATEFAILED), string(platform.ClusterStatusCREATEFAILED)},
+		Refresh:    r.resourceRefreshFunc(ctx, cluster.JSON200.Id),
+		Timeout:    3 * time.Hour,
```

Comment:
```suggestion
		Timeout:    1 * time.Hour,
```
since just updating workspaces, it should be much shorter